content/200-concepts/100-components/02-prisma-client/053-client-extensions/140-shared-extensions.mdx

```diff
@@ -7,50 +7,92 @@ tocDepth: 4
 
 <TopBlock>
 
-You can share your [Prisma Client extensions](/concepts/components/prisma-client/client-extensions) with other users, either as packages or as code snippets, and import extensions that other users create into your Prisma projects.
+You can share your [Prisma Client extensions](/concepts/components/prisma-client/client-extensions) with other users, either as packages or as modules, and import extensions that other users create into your project.
+
+If you would like to build a shareable extension, we also recommend using the [`prisma-client-extension-starter`](https://github.com/prisma/prisma-client-extension-starter) template.
 
 </TopBlock>
 
 ## Share Prisma Client extensions
 
-When you want to create extensions that will work for other users, and not just for your specific schema, use the `$all` prefix, such as [`$allModels`](/concepts/components/prisma-client/client-extensions/model#add-a-custom-method-to-all-models-in-your-schema). Extensions that use `$all` apply generally, not just to a named component of a specific schema.
+When you want to create extensions other users can use, and that are not tailored just for your schema, Prisma provides utilities to allow you to create shareable extensions.
+
+### Define an extension
+
+Use the `Prisma.defineExtension` method to make your extension shareable. You can use it to package the extension to either separate your extensions into a separate file or share it with other users as an npm package.
+
+The benefit of `Prisma.defineExtension` is that it provides strict type checks and auto completion for authors of extension in development and users of shared extensions.
 
-You do not need to use the `$all` prefix with the [`client`](/concepts/components/prisma-client/client-extensions/client) component, because the `client` component always applies generally.
+To define an extension:
+
+1. Define the extension as a module using `Prisma.defineExtension`
+2. Use one of the methods that begin with the `$all` prefix such as [`$allModels`](/concepts/components/prisma-client/client-extensions/model#add-a-custom-method-to-all-models-in-your-schema) or `$allOperations`
+
+Extensions that contain methods with the `$all` prefix apply to a client instance and not to a named component, e.g. `result` or `query`, of a specific instance.
```

Comment:
```suggestion
Extensions that contain methods under `$allModels` apply to every model instead of a specific one. Similarly, methods under `$allOperations` apply to a client instance as a whole and not to a named component, e.g. `result` or `query`.
```